src/OmniSharp.Roslyn/Extensions/SolutionExtensions.cs

```diff
@@ -86,7 +90,25 @@ private static QuickFix ConvertSymbol(Solution solution, ISymbol symbol, Locatio
                 Column = lineSpan.StartLinePosition.Character,
                 EndLine = lineSpan.EndLinePosition.Line,
                 EndColumn = lineSpan.EndLinePosition.Character,
-                Projects = projects
+                Projects = projects,
+                ContainingSymbolName = symbol.ContainingSymbol?.Name ?? "",
+                GeneratedFileInfo = GetSourceGeneratedFileInfo(solution, location),
+            };
+        }
+
+        internal static SourceGeneratedFileInfo? GetSourceGeneratedFileInfo(Solution solution, Location location)
```

Comment:
Copy/paste from `GoToDefinitionHelpers`, and modified to take a `Solution` instead of an `OmniSharpWorkspace`.

Reply:
I believe this was intended to be the home for extension methods, so maybe it would be good to make this one too for consistency